File: data_structures/stack/stack.go
package main

import "fmt"

type Stack struct {
	values []int
}

func (s *Stack) Push(item int) {
	s.values = append(s.values, item)
}

func (s *Stack) Pop() int {
	lastIndex := len(s.values) - 1
	toRemove := s.values[lastIndex]
	s.values = s.values[:lastIndex]
	return toRemove
}

func main() {
	stack := Stack{}
	fmt.Println(stack)
	stack.Push(10)
	stack.Push(20)
	stack.Push(30)
	stack.Pop()

	fmt.Println(stack)
}

File: data_structures/stack/stack.py
class Stack:
    def __init__(self):
        self.data = []
    
    def push(self, el):
        self.data.append(el)
    
    def pop(self):
        return self.data.pop()
    
    def peek(self):
        return self.data[-1]
    
    def is_empty(self):
        return len(self.data) == 0


stack = Stack()

stack.push(1)
stack.push(2)
stack.push(3)
stack.push(4)
stack.push(5)

print(stack.pop())
print(stack.peek())

print(stack.is_empty())
